src/app/profile/views/ens_manager.nim

```diff
@@ -203,18 +200,33 @@ QtObject:
     self.endResetModel()
     self.transactionCompleted(false, transactionHash, ensUsername, $trxType, revertReason)
 
-  proc registerENS(self: EnsManager, username: string, password: string) {.slot.} =
-    let pubKey = status_settings.getSetting[string](Setting.PublicKey, "0x0")
-    let address = status_wallet.getWalletAccounts()[0].address
-    let walletAddress = parseAddress(address)
-    let trxHash = registerUsername(username, walletAddress, pubKey, password)
-    
-    self.transactionWasSent(trxHash)
+  proc getEnsRegisterAddress(self: EnsManager): QVariant {.slot.} =
+    newQVariant($statusRegistrarAddress())
 
-     # TODO: handle transaction failure
-    var ensUsername = formatUsername(username, true)
-    self.pendingUsernames.incl(ensUsername)
-    self.add ensUsername
+  QtProperty[QVariant] ensRegisterAddress:
+    read = getEnsRegisterAddress
+
+  proc registerENSGasEstimate(self: EnsManager, ensUsername: string, address: string): int {.slot.} =
+    let pubKey = status_settings.getSetting[string](Setting.PublicKey, "0x0")
+    try:
+      result = registerUsernameEstimateGas(ensUsername, address, pubKey)
+    except:
+      result = 325000
+  
+  proc registerENS*(self: EnsManager, username: string, address: string, gas: string, gasPrice: string, password: string): string {.slot.} =
+    try:
+      let pubKey = status_settings.getSetting[string](Setting.PublicKey, "0x0")
+      let response = registerUsername(username, pubKey, address, gas, gasPrice, password)
+      result = $(%* { "result": %response })
+      self.transactionWasSent(response)
+
+      # TODO: handle transaction failure
```

Comment:
Assuming this is updated to the error handling spec'd in the PR that allows for error handling in threads, then this could be done pretty easily, just like this: https://github.com/status-im/nim-status-client/pull/918/files#diff-ed6f2995c8c8aa27163e93371ed67be4R121-R122